### PR TITLE
Clarify terminal is child window of details panel.

### LINKED ISSUE #1152: it's not obvious that terminals are extensions of the details panel
I only discovered by accident that closing the details panel also closes any associated terminal panel - up until that point I had been dutifully closing the latter and then the former.

Can we make the relationship between the two more obvious?

Perhaps the terminal could look like (and reveal like) a side pull-out from the details panel?

@peterbourgon says such ideas were discussed before. Might be worth revisiting.

### PR DESCRIPTION
**Pull request by foot**
So when you close the details panel and the terminal closes too its not a surprise!

Fixes #1152

### CODE REVIEW

**Comment by rade** [comment]
screen cap?

**Comment by rade** [comment]
any chance we could change the 'close' to a 'slide in'?

**Comment by rade** [comment]
and perhaps change the icon too, e.g. ">>"

**Comment by foot** [comment]
`>>` / `slide in` suggests some kind of permanence/continuity to me, as in, _I'm done with this terminal session for now, lets tuck it away for later._

I like: boom, its gone, gotta make a new one now.

Though you could interpret the idea of the having access to the container as continuous thing here and it would make more sense summoning it back, thoughts? (+ @davkal)

**Comment by rade** [comment]
I see your point.

My concern is that I see the "x" button and "boom, its gone" action as indicators that the terminal has an independent life from the detail panel.

**Comment by foot** [comment]
They feel quite connected now.

There are other onClose animations we could look into but I don't wanna make it too animation heavy, and I think a snappy close is usually nice: Done here, I'm off to do something else. Vs establishing context etc as we show something new.

Alt close animation: 
 - Keep `x`
 - fade out + sliding back, signifying destruction + going back home to its parent the details panel.

**Comment by davkal** [comment]
I'm happy with the immediate close on X. 

screen shot 2016-10-18 at 14 46 30

Not so happy with the aesthetic. Both panels look a bit bolted together, colors are clashing a bit. There is no clear hierarchy (other than suggested by the animation). Most oddly, the details panel has a shadow, but the terminal has not.

Counter-proposal: remove the dark color border, add padding (to make it smaller to suggest hierarchy), and add shadow.

screen shot 2016-10-18 at 14 46 12

File: client/app/scripts/components/terminal.js [hunk 1]
```diff
@@ -328,4 +353,10 @@ class Terminal extends React.Component {
   }
 }
 
+
+Terminal.defaultProps = {
+  connect: true
```
Comment (davkal) [inline client/app/scripts/components/terminal.js]: What's this for?

File: client/app/styles/main.less [hunk 2]
```diff
@@ -979,35 +980,34 @@ h2 {
...
-    right: (@details-window-width + @details-window-padding-left + 10px);
+    position: relative;
+    flex: 1;
+    overflow-x: hidden;
+  }
+
+  &-animation-wrapper {
+    width: 100%;
+    height: 100%;
+    transition: transform 0.5s cubic-bezier(0.230, 1.000, 0.320, 1.000);
```
Comment (davkal) [inline client/app/styles/main.less]: Maybe extract transform timing function?

File: client/app/scripts/components/terminal.js (hunk 1 above)
Comment (foot) [inline client/app/scripts/components/terminal.js]: Terminal doesn't try and connect until it receives the connect attr.
`<Terminal connect />`

We don't want to connect until after the drawer animation has completed. Term.js hits the dom a lot and makes the animation super janky.

**Comment by davkal** [comment]
YES! I like the hover bit. It feels like the details panel owns the terminal. Well done. LGTM

**Pull request merged**